### ISSUE TITLE
python header file parser can't handle line breaks in functions with va_list args

### ISSUE THREAD

**Issue by garlick** [flux-framework/flux-core#1221]
File this under "there has got to be a better way".

I just added a public KVS function that has a va_list argument, and found the python wrapper generator blew up with
```
/usr/bin/python ../../../../src/bindings/python/make_binding.py --path ../../../../src/common/libkvs \
  			  --package flux \
                          --modname _kvs \
			  --include_ffi _core_build \
			  kvs.h
/usr/bin/python _kvs_build.py
Traceback (most recent call last):
  File "_kvs_build.py", line 210, in <module>
    """)
  File "/usr/lib/python2.7/dist-packages/cffi/api.py", line 105, in cdef
    self._cdef(csource, override=override, packed=packed)
  File "/usr/lib/python2.7/dist-packages/cffi/api.py", line 119, in _cdef
    self._parser.parse(csource, override=override, **options)
  File "/usr/lib/python2.7/dist-packages/cffi/cparser.py", line 289, in parse
    self._internal_parse(csource)
  File "/usr/lib/python2.7/dist-packages/cffi/cparser.py", line 294, in _internal_parse
    ast, macros, csource = self._parse(csource)
  File "/usr/lib/python2.7/dist-packages/cffi/cparser.py", line 252, in _parse
    self.convert_pycparser_error(e, csource)
  File "/usr/lib/python2.7/dist-packages/cffi/cparser.py", line 281, in convert_pycparser_error
    raise api.CDefError(msg)
cffi.api.CDefError: cannot parse "const char *key, const char *fmt, va_list ap);"
:167:59: before: va_list
Makefile:1024: recipe for target '_kvs.c' failed
make[2]: *** [_kvs.c] Error 1

```

In `src/bindings/python/flux/Makefile.am` there is a rule to build `_core_build.py` that looks like this:
```
_core_build.py: $(MAKE_BINDING)
        $(PYTHON) $(MAKE_BINDING) --path $(top_srcdir)/src/common/libflux \
                                  --package flux \
                                  --modname _core \
                                  --add_sub '.*va_list.*|||' \
                                  --ignore_header 'handle_impl' \
                                  --ignore_header 'macros' \
                                  --add_long_sub 'FLUX_SEC_TYPE_ALL.*\n.*\),|||FLUX_SEC_TYPE_ALL = 7,'\
                                  flux.h
```
So I replicated the `--add_sub` argument for va_list in the rule to build `_kvs_build.py`, but this also failed with
```
/usr/bin/python ../../../../src/bindings/python/make_binding.py --path ../../../../src/common/libkvs \
  			  --package flux \
                          --modname _kvs \
			  --add_sub '.*va_list.*|||' \
			  --include_ffi _core_build \
			  kvs.h
/usr/bin/python _kvs_build.py
Traceback (most recent call last):
  File "_kvs_build.py", line 209, in <module>
    """)
  File "/usr/lib/python2.7/dist-packages/cffi/api.py", line 105, in cdef
    self._cdef(csource, override=override, packed=packed)
  File "/usr/lib/python2.7/dist-packages/cffi/api.py", line 119, in _cdef
    self._parser.parse(csource, override=override, **options)
  File "/usr/lib/python2.7/dist-packages/cffi/cparser.py", line 289, in parse
    self._internal_parse(csource)
  File "/usr/lib/python2.7/dist-packages/cffi/cparser.py", line 294, in _internal_parse
    ast, macros, csource = self._parse(csource)
  File "/usr/lib/python2.7/dist-packages/cffi/cparser.py", line 252, in _parse
    self.convert_pycparser_error(e, csource)
  File "/usr/lib/python2.7/dist-packages/cffi/cparser.py", line 281, in convert_pycparser_error
    raise api.CDefError(msg)
cffi.api.CDefError: cannot parse "const char *key, const char *fmt,  __dotdotdot__ );"
:169:74: before: ;
Makefile:1024: recipe for target '_kvs.c' failed
make[3]: *** [_kvs.c] Error 1
```

Ultimately I got it working by removing a line break from my function prototype and letting it run beyond 80 columns.

**Comment by garlick** [comment]
Adding ref to #1223 in which this script was improved somewhat.

**Comment by garlick** [comment]
My guess is that the problem is how lines are split in `make_binding.py::process_header()`.

We're calling the script with `--add_sub '.*va_list.*|||'` and only the part of the function definition containing va_list is being dropped, leaving the continuation line with half a function definition there to confuse cffi.

Probably a straightforward fix  for someone with python skills - @morrone maybe?

**Comment by SteVwonder** [comment]
@trws: this was fixed in #1795, right? Can we close this?

**Comment by trws** [comment]
Closing is reasonable, but the fix is heuristic rather than certain.  If anyone
declares a function pointer's type inline, it will still cause issues.
